lib/cart.js

```diff
@@ -0,0 +1,56 @@
+import { useState, useCallback } from 'react'
+import useSWR, { mutate } from 'swr'
+
+async function getError(res) {
+  if (res.headers.get('Content-Type').includes('application/json')) {
+    const data = await res.json()
+    return data.errors[0]
+  }
+  return { message: (await res.text()) || res.statusText }
+}
+
+async function fetcher(url) {
+  const res = await fetch(url)
+
+  if (res.status === 200) {
+    return res.json()
+  }
+  throw await getError(res)
+}
+
+export function useCart() {
+  return useSWR('/api/cart', fetcher)
+}
```

Comment:
This is currently not very useful, but allows for an easier usage of the cart's data, and also allows for more features in the future.